runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -48,16 +51,78 @@ class MM_ObjectAllocationAPI
 	 * Data members
 	 */
 private:
-	const UDATA _gcAllocationType;
+	const uintptr_t _gcAllocationType;
 #if defined(J9VM_GC_BATCH_CLEAR_TLH)
-	const UDATA _initializeSlotsOnTLHAllocate;
+	const uintptr_t _initializeSlotsOnTLHAllocate;
 #endif /* J9VM_GC_BATCH_CLEAR_TLH */
-	const UDATA _objectAlignmentInBytes;
+	const uintptr_t _objectAlignmentInBytes;
 
 #if defined (J9VM_GC_SEGREGATED_HEAP)
 	const J9VMGCSizeClasses *_sizeClasses;
 #endif /* J9VM_GC_SEGREGATED_HEAP */
 
+	VMINLINE void
+	initializeIndexableSlots(bool initializeSlots, uintptr_t dataSize, void *dataAddr)
+	{
+		if ((NULL != dataAddr) && initializeSlots) {
```

Comment:
(NULL != dataAddr) is just temporary till all parties do initialize/update dataAddr? Even in 0-sized arrays, this will be initialized (to the end of header)?

Reply:
Yes, `dataAddr` will always be initialized to:
```
void *dataAddr = (void *)((uintptr_t)header + headerSize);
```
`(NULL != dataAddr)` is there for safety even though it's not necessary since the default is to init `dataAddr` to point to data right after header.